src/Disks/IO/ReadBufferFromAzureBlobStorage.cpp

```diff
@@ -223,12 +203,18 @@ void ReadBufferFromAzureBlobStorage::initialize()
         sleep_time_with_backoff_milliseconds *= 2;
     };
 
+    long read_bytes = 0;
+
     for (size_t i = 0; i < max_single_download_retries; ++i)
     {
         try
         {
-            auto download_response = blob_client->Download(download_options);
-            data_stream = std::move(download_response.Value.BodyStream);
+            auto download_response = blob_client->DownloadTo(reinterpret_cast<uint8_t *>(data_ptr), data_capacity, download_options);
```

Comment:
I think it will not improve performance but will decrease it instead because it will make a GET request for each call of `nextImpl` instead of one request in `initialize` method, as I understand the azure sdk.